### ISSUE TITLE
Readme doesn't describe priority order for opt-out features

### ISSUE THREAD

**Issue by Hixie** [ossf/allstar#61]
There are various ways to opt repos in and out, including these:

```
optConfig:
  optOutStrategy: true
  optOutRepos:
  - ...
  optInRepos:
  - ...
  optOutPrivateRepos: true
  optOutPublicRepos: false
```
...and including a per-repo file and an org-wide .allstar repo file.

It is highly unclear from the documentation what will win if there is a conflict ~~(additionally it seems there's an undocumented feature that changes which will win, as mentioned briefly in opt-out.md, which makes this even more unclear)~~. (edit: found the documentation for this feature)

For example, does `optOutPrivateRepos` win vs an explicit `optInRepos`? What if `optOutStrategy` is true, but `optOutPrivateRepos` and `optOutPublicRepos` are also true? What if a repo opts out locally but is listed explicitly in the global .allstar file?

**Comment by jeffmendoza** [comment]
Ignoring `optOutPrivateRepos` and `optOutPublicRepos`:
Firstly, there are two "modes" or "strategies" that Allstar can be used in to enable repos: opt-in or opt-out, This is configured using the `optOutStrategy` setting.

In opt-in strategy (`optOutStrategy`=`false`, default), The only other setting used is `optInRepos`. Those listed will be opted-in and enabled. `optOutRepos` is ignored. If repo-override is allowed. Repos can opt-in with settings in the repo as well, opt-out settings are ignored.

In opt-out strategy (`optOutStrategy`=`true`), `optOutRepos` is used. All repos are enabled, but those listed in `optOutRepos` are opted-out. Additionally, if repo-override is allowed, repos can opt-out with the setting in their own repo file. All the opt-in settings are ignored.

Back to `optOutPrivateRepos` and `optOutPublicRepos`, these only work in opt-out strategy. They opt-out all public/private repos if set to true. These were recently added and break the symmetry. Please advise if you think these would be useful in opt-in mode as well.

Sorry for the confusion!

**Comment by Hixie** [comment]
This would be great information to add to the readme. :-)

**Comment by olivekl** [comment]
Fixed by #84.
